spec.bs

```diff
@@ -1819,6 +1826,55 @@ Several APIs specific to fenced frames are defined on the {{Fence}} interface.
   </wpt>
 </div>
 
+<div algorithm>
+  The <dfn method for=Fence>disableUntrustedNetwork()</dfn> method steps are:
+
+  1. Let |p| be [=a new promise=].
+
+  1. Let |instance| be [=this=]'s [=relevant global object=]'s [=Window/browsing context=]'s
+     [=browsing context/fenced frame config instance=].
+
+  1. If |instance| is null, then [=resolve=] |p| with {{undefined}} and return |p|.
+
+  1. [=Assert=]: the [=relevant settings object=]'s [=environment settings object/origin=] and
+     |instance|'s [=fenced frame config instance/mapped url=]'s [=url/origin=] are [=same origin=].
+
+     Note: Only contexts that are [=same origin=] to the [=fenced frame config instance/mapped url=]
+     described in the [=fenced frame config instance=] are allowed to call this API.
```

Comment:
I'm happy the assert is "sensible" and passes, but I still don't quite understand it. Is it even possible for same-origin but different-from-the-original-navigation Documents to make it here? Or would the config be reset for those navigations?

Reply:
Actually, this should be a promise rejection with a TypeError (rather than an undefined resolution) to match what we do in impl. Since cross-origin subframes get access to `window.fence`, they are able to call this method.

It is possible if a same-origin child iframe calls `disableUntrustedNetwork()`, which is a case that we allow. It's also possible if the fenced frame performs a same-origin navigation before having the new navigated document call `disableUntrustedNetwork()`.